src/FESpaces/FESpacesWithDofPotentiallyRemoved.jl

```diff
@@ -0,0 +1,185 @@
+"""
+    struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
+      space::SingleFieldFESpace
+      constraint_style::Val{CS}
+      remove_dof::Val{RD}
+      dof_to_remove::Int
+    end
+"""
+struct FESpaceWithDofPotentiallyRemoved{CS,RD} <: SingleFieldFESpace
+  space::SingleFieldFESpace
+  constraint_style::Val{CS}
+  remove_dof::Val{RD}
+  dof_to_remove::Int
+  @doc """
+      FESpaceWithDofPotentiallyRemoved(space::SingleFieldFESpace, remove_dof::Bool,
+      dof_to_remove::Int=num_free_dofs(space))
+  """
+  function FESpaceWithDofPotentiallyRemoved(space::SingleFieldFESpace,
+     remove_dof::Bool,
+     dof_to_remove::Int=num_free_dofs(space))
+    s = "FESpaceWithDofPotentiallyRemoved can only be constructed from spaces without dirichlet dofs."
```

Comment:
Probably do nothing in this case. It reminds me Balancing Neumann-Neumann local spaces. If the constant is already fixed, do nothing. Minor point.

Reply:
Done in 0da971d.